saas-starter-ts/backend/subscription/stripe.ts

```diff
@@ -0,0 +1,258 @@
+  return new Promise((resolve) => {
+    const bodyParts: Uint8Array[] = [];
+    req
+      .on("data", (chunk) => {
+        bodyParts.push(chunk);
+      })
+      .on("end", () => {
+        resolve(Buffer.concat(bodyParts).toString());
+      });
+  });
+}
+
+// Handle stripe webhook events
+export const stripeWebhookHandler = api.raw(
+  {
+    method: "POST",
+    expose: true,
+    path: "/stripe/webhook",
+  },
+  async (req, res) => {
+    try {
+      const body = await getBody(req);
+      const stripeSignature = req.headers["stripe-signature"] as string;
+
+      const event = stripe.webhooks.constructEvent(
+        body,
+        stripeSignature,
+        stripeWebhookSigningSecret(),
+      );
+
+      log.debug("received stripe webhook", { event });
+
+      switch (event.type) {
+        // Handle subscription events
+        case "customer.subscription.updated":
+        case "customer.subscription.created": {
+          const subscription = event.data.object;
+          await db.exec`
+        INSERT INTO subscription (stripe_id, stripe_customer_id, price_id, status, created_at, updated_at)
+        VALUES (
+          ${subscription.id as string},
+          ${subscription.customer as string},
+          ${subscription.items.data[0].plan.id as string},
+          ${subscription.status as string},
+          ${new Date(subscription.created * 1000)},
+          ${new Date(subscription.created * 1000)}
+        )
+        ON CONFLICT (stripe_id) DO UPDATE SET
+          stripe_customer_id = ${subscription.customer as string},
+          price_id = ${subscription.items.data[0].plan.id as string},
+          status = ${subscription.status as string},
+          updated_at = ${new Date(subscription.created * 1000)}
+        `;
+          break;
+        }
+      }
+
+      res.write(JSON.stringify({ received: true }));
+      res.writeHead(200);
+      res.end();
+    } catch (error) {
+      log.error("stripe webhook error", { error });
+
+      res.write(JSON.stringify({ received: false }));
+      res.writeHead(500);
+      res.end();
+    }
+  },
+);
+
+interface CreateCheckoutSessionParams {
+  priceId: string;
+}
+
+interface CreateCheckoutSessionResponse {
+  // The URL to redirect to
+  url: string;
+}
+
+// Generate a checkout session
+export const createCheckoutSession = api(
+  {
+    method: "POST",
+    path: "/stripe/checkout-session",
+    expose: true,
+    auth: true,
+  },
+  async (
+    params: CreateCheckoutSessionParams,
+  ): Promise<CreateCheckoutSessionResponse> => {
+    const authData = getAuthData();
+    if (!authData) {
+      throw APIError.unauthenticated("user not authenticated");
+    }
+
+    // Try to get the organization's stripe customer ID
+    // This is stored as organization metadata, please see https://clerk.com/docs/organizations/metadata for more information.
+    const org = await clerk.organizations.getOrganization({
+      organizationId: authData.orgID,
```

Comment:
Why are we guaranteed this is non-null? It was coerced into an empty string in the auth handler when missing